libraries/libcss/src/main/java/org/pentaho/reporting/libraries/css/model/StyleKeyRegistry.java

```diff
@@ -37,7 +37,7 @@
   private static final Log logger = LogFactory.getLog( StyleKeyRegistry.class );
   private static StyleKeyRegistry registry;
 
-  public static synchronized StyleKeyRegistry getRegistry() {
+  public static StyleKeyRegistry getRegistry() {
```

Comment:
You will need a form of memory barrier here or you might end up with stale data. Booting will set the registry, and if someone queried the registry on a different thread before that happens they might not see the updated instance after booting until they hit a syncronized block elsewhere. Make the registry variable either volatile or use a atomic reference to make this safe.

Reply:
Incorporated feedback with a ReentrantReadWriteLock, and squashed commits.

Reply:
Fixed with a ReentrantReadWriteLock.